runtime/lua/vim/treesitter.lua

```diff
@@ -241,4 +241,223 @@ function M.get_captures_at_position(bufnr, row, col)
   return matches
 end
 
+--- Gets the smallest named node under the cursor
+---
+---@param winnr number Window handle or 0 for current window
+---@param opts table Options table
+---@param opts.ignore_injections boolean (default true) Ignore injected languages.
+---
+---@returns (table) The named node under the cursor
+function M.get_node_at_cursor(winnr, opts)
+  winnr = winnr or 0
+  local cursor = a.nvim_win_get_cursor(winnr)
+  local ts_cursor_range = { cursor[1] - 1, cursor[2], cursor[1] - 1, cursor[2] }
+
+  local buf = a.nvim_win_get_buf(winnr)
+  local root_lang_tree = M.get_parser(buf)
+  if not root_lang_tree then
+    return
+  end
+
+  return root_lang_tree:named_node_for_range(ts_cursor_range, opts)
+end
+
+---Gets a compatible vim range (1 index based) from a TS node range.
+---
+---TS nodes start with 0 and the end col is ending exclusive.
+---They also treat a EOF/EOL char as a char ending in the first
+---col of the next row.
+---
+---@param bufnr number The buffer handle from which the range is from
+---@param range table The treesitter range to transform
+---
+---@returns start_row, starcol, end_row, end_col
+function M.get_vim_range(bufnr, range)
+  local srow, scol, erow, ecol = unpack(range)
+  srow = srow + 1
+  scol = scol + 1
+  erow = erow + 1
+
+  if ecol == 0 then
+    -- Use the value of the last col of the previous row instead.
+    erow = erow - 1
+    if not bufnr or bufnr == 0 then
+      ecol = vim.fn.col({ erow, '$' }) - 1
+    else
+      ecol = #a.nvim_buf_get_lines(bufnr, erow - 1, erow, false)[1]
+    end
+  end
+  return srow, scol, erow, ecol
+end
+
+---Highlights the given range
+---
+---@param bufnr number The buffer number
+---@param range table The treesitter range to highlight
+---@param ns number The namespace id
+---@param hlgroup string The highlight group name
+function M.highlight_range(bufnr, range, ns, hlgroup)
+  local start_row, start_col, end_row, end_col = unpack(range)
+  vim.highlight.range(bufnr, ns, hlgroup, { start_row, start_col }, { end_row, end_col })
```

Comment:
```suggestion
  vim.highlight.range(bufnr, ns, hlgroup, { start_row, start_col }, { end_row, end_col }, { priority = vim.highlight.treesitter })
```

Reply:
The idea was actually to use this function in user code and thus have the normal higher user priority to paint over tree-sitter. Basically to import a tree-sitter range for non-tree-sitter highlights. Priority could go into an opts argument

Reply:
Makes sense. Let's add the priority `opts` then and document that it defaults to `user`.